src/App/View/LinkedText.purs

```diff
@@ -2,63 +2,94 @@ module App.View.LinkedText where
+import Data.Tuple (fst, snd)
+import Effect (Effect)
+import Effect.Class.Console (logShow)
+import Effect.Console (log)
+import Primitive (string, unpack)
+import Util ((!))
 import Val (Val)
+import Web.Event.EventTarget (EventListener)
 
-foreign import drawLinkedText :: LinkedTextHelpers -> Renderer LinkedText
+newtype LinkedText = LinkedText (Array (Selectable String))
 
-type LinkedTextHelpers =
-   { explanation :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   , contents :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   , accessAnn :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> SelState 𝕊
-   }
+contents :: Selectable String -> String
+contents = fst
 
-newtype LinkedText = LinkedText (Array (Selectable String + Selectable (Explanation (SelState 𝕊))))
-
-drawLinkedText' :: Renderer LinkedText
-drawLinkedText' = drawLinkedText linkedTextHelpers
-
-linkedTextHelpers :: LinkedTextHelpers
-linkedTextHelpers =
-   { explanation
-   , contents
-   , accessAnn
-   }
-   where
-   explanation :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   explanation = case _ of
-      Left s -> fst s
-      Right (Explanation _ expl _ × _) -> expl
-
-   contents :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   contents = case _ of
-      Left s -> fst s
-      Right (Explanation _ _ v × _) -> fst (unpack string v)
-
-   accessAnn :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> SelState 𝕊
-   accessAnn = case _ of
-      Left (_ × α) -> α
-      Right (Explanation α _ _ × _) -> α
+accessAnn :: Selectable String -> SelState 𝕊
+accessAnn = snd
 
 instance Drawable LinkedText where
    draw rSpec figVal _ redraw =
-      drawLinkedText linkedTextHelpers uiHelpers rSpec =<< selListener figVal redraw linkedTextSelector
+      draw' uiHelpers rSpec =<< selListener figVal redraw linkedTextSelector
       where
       linkedTextSelector :: ViewSelSetter LinkedTextElem
       linkedTextSelector { i } = linkedText <<< listElement i
 
-exch :: forall a. Tuple (Either String (Explanation a)) a -> Either (Tuple String a) (Tuple (Explanation a) a)
-exch (e × a) = case e of
-   Left s -> Left (s × a)
-   Right ex -> Right (ex × a)
+setSelState :: LinkedText -> EventListener -> D3.Selection -> Effect Unit
+setSelState (LinkedText elems) redraw rootElement = do
+   elems' <- rootElement # selectAll ".linked-text"
+   logShow (length elems')
+   for_ elems' \elem -> do
+      elem' :: LinkedTextElem <- datum elem
+      log "About to set attrs"
+      log $ "Elem': " <> (show elem')
+      elem # setStyles (textAttrs elem') >>= registerMouseListeners redraw
+      log "set attrs"
+   log "ok"
+   where
+   textAttrs :: LinkedTextElem -> Attrs
+   textAttrs { i } =
+      [ "border" ↦ border
+      , "background" ↦ background
+      , "color" ↦ color
+      ]
+      where
+      sel = accessAnn (elems ! i)
+
+      border :: String
+      border
```

Comment:
I think we need to revert to underlining only so that the text doesn't move when transiently highlighted.